packages/frontend/apps/ios/App/App/plugins/Cookie/HashcashPlugin.swift

```diff
@@ -0,0 +1,107 @@
+import Capacitor
+import CryptoSwift
+
+@objc(HashcashPlugin)
+public class HashcashPlugin: CAPPlugin, CAPBridgedPlugin {
+    public let identifier = "HashcashPlugin"
+    public let jsName = "Hashcash"
+    public let pluginMethods: [CAPPluginMethod] = [
+        CAPPluginMethod(name: "hash", returnType: CAPPluginReturnPromise)
+    ]
+
+    @objc func hash(_ call: CAPPluginCall) {
+        DispatchQueue.global(qos: .default).async {
+          let challenge = call.getString("challenge") ?? ""
+          let bits = call.getInt("bits") ?? 20;
+          call.resolve(["value": Stamp.mint(resource: challenge, bits: UInt32(bits)).format()])
+        }
+    }
+}
+
+let SALT_LENGTH = 16
+
+struct Stamp {
+    let version: String
+    let claim: UInt32
+    let ts: String
+    let resource: String
+    let ext: String
+    let rand: String
+    let counter: String
+    
+    func checkExpiration() -> Bool {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyyMMddHHmmss"
+        guard let date = dateFormatter.date(from: ts) else { return false }
+        return Date().addingTimeInterval(5 * 60) <= date
+    }
+    
+    func check(bits: UInt32, resource: String) -> Bool {
+        if version == "1" && bits <= claim && checkExpiration() && self.resource == resource {
+            let hexDigits = Int(floor(Float(claim) / 4.0))
+            
+            // Check challenge
+            let formatted = format()
+            var sha3256 = SHA3.init(variant: .sha256)
+            let result = formatted.data(using: .utf8)!.sha3(.sha256).compactMap { String(format: "%02x", $0) }.joined()
+            return result.prefix(hexDigits) == String(repeating: "0", count: hexDigits)
```

Comment:
it looks a bit strange, a sha3 instance is created here but not used, instead the hash is generated through .data.sha3